include/triton/core/tritonrepoagent.h

```diff
@@ -0,0 +1,410 @@
+///     'location' returns the full path to the remote directory that contains
+///     the model's artifacts. The returned location string is owned by Triton,
+///     not the caller, and so should not be modified or freed. The contents of
+///     the remote directory are owned by Triton, not the agent,
+///     and so the agent should not delete or modify the contents.
+///     Use TRITONREPOAGENT_ModelRepositoryLocationAcquire to get a location
+///     that can be used to write updated model repository contents.
+///
+/// \param agent The agent.
+/// \param model The model.
+/// \param artifact_type Returns the artifact type for the location.
+/// \param path Returns the location.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_DECLSPEC TRITONSERVER_Error*
+TRITONREPOAGENT_ModelRepositoryLocation(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model,
+    TRITONREPOAGENT_ArtifactType* artifact_type, const char** location);
+
+/// Acquire a location where the agent can produce a new version of
+/// the model repository files. This is a convenience method to create
+/// a temporary directory for the agent. The agent is responsible for
+/// calling TRITONREPOAGENT_ModelRepositoryLocationDelete in
+/// TRITONREPOAGENT_ModelFinalize to delete the location. Initially the
+/// acquired location is empty. The 'location' communicated depends on
+/// the requested 'artifact_type'.
+///
+///   TRITONREPOAGENT_ARTIFACT_FILESYSTEM: The location is a directory
+///     on the local filesystem. 'location' returns the full path to
+///     an empty directory that the agent should populate with the
+///     model's artifacts. The returned location string is owned by
+///     Triton, not the agent, and so should not be modified or freed.
+///
+/// \param agent The agent.
+/// \param model The model.
+/// \param artifact_type The artifact type for the location.
+/// \param path Returns the location.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_DECLSPEC TRITONSERVER_Error*
+TRITONREPOAGENT_ModelRepositoryLocationAcquire(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model,
+    const TRITONREPOAGENT_ArtifactType artifact_type, const char** location);
+
+/// Discard and release ownership of a previously acquired location
+/// and its contents. The agent must not access or modify the location
+/// or its contents after this call.
+///
+/// \param agent The agent.
+/// \param model The model.
+/// \param path The location to release.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_DECLSPEC TRITONSERVER_Error*
+TRITONREPOAGENT_ModelRepositoryLocationDelete(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model,
+    const char* location);
+
+/// Inform Triton that the specified repository location should be used for
+/// the model in place of the original model repository. The 'location'
```

Comment:
I forgot to mention that we only want this method called when in the LOAD state and Triton should enforce this. 
"... model repository. This method can only be called when TRITONREPOAGENT_ModelAction is invoked with TRITONREPOAGENT_ACTION_LOAD. The 'location' ..."